lib/smack/SmackRep.cpp

```diff
@@ -98,6 +171,16 @@ string SmackRep::memReg(unsigned idx) {
   return s.str();
 }
 
+string SmackRep::memReg(unsigned idx, bool safety, unsigned type) {
```

Comment:
This `memReg` combines the region number with the access type (e.g. `bytes`, `i8`, `i16`, …). It would be cleaner to separate these two, having `memReg` returning only `"$M.X"` and concatenating the result with `memType` when necessary. Furthermore, is there any reason to have separate types `bytes` and `i8`?